corehq/apps/domain/views/settings.py

```diff
@@ -517,3 +520,87 @@ class ManageDomainMobileWorkersView(ManageMobileWorkersMixin, BaseAdminProjectSe
     page_title = gettext_lazy("Manage Mobile Workers")
     template_name = 'enterprise/manage_mobile_workers.html'
     urlname = 'domain_manage_mobile_workers'
+
+
+@method_decorator(toggles.CUSTOM_DOMAIN_BANNER_ALERTS.required_decorator(), name='dispatch')
+class ManageDomainAlertsView(BaseAdminProjectSettingsView):
+    template_name = 'domain/admin/manage_alerts.html'
+    urlname = 'domain_manage_alerts'
+    page_title = gettext_lazy("Manage Project Alerts")
+
+    @property
+    def page_context(self):
+        return {
+            'form': self.form,
+            'alerts': [
+                {
+                    'active': alert.active,
+                    'html': alert.html,
+                    'id': alert.id,
+                    'created_by_user': alert.created_by_user,
+                }
+                for alert in CommCareHQAlert.objects.filter(created_by_domain=self.domain)
+            ]
+        }
+
+    @cached_property
+    def form(self):
+        if self.request.method == 'POST':
+            return DomainAlertForm(self.request.POST)
+        return DomainAlertForm()
+
+    def post(self, request, *args, **kwargs):
+        command = request.POST.get('command')
+        if command:
+            self._apply_command(request, command)
+        elif self.form.is_valid():
+            self._create_alert()
+            messages.success(request, _("Alert saved!"))
+        else:
+            messages.error(request, _("There was an error saving your alert. Please try again!"))
+            return self.get(request, *args, **kwargs)
+        return HttpResponseRedirect(self.page_url)
+
+    def _apply_command(self, request, command):
+        alert_id = request.POST.get('alert_id')
+        if alert_id:
+            if command == 'delete':
+                self._delete_alert(alert_id)
+            else:
+                success = self._update_alert(alert_id, command)
+                if success:
+                    messages.success(request, _("Alert updated!"))
+                else:
+                    messages.error(request, _("Could not update alert. Please try again!"))
+
+    def _update_alert(self, alert_id, command):
+        alert = self._load_alert(alert_id)
+        if alert:
+            if command == 'activate':
+                alert.active = True
+            elif command == 'deactivate':
+                alert.active = False
+            alert.save(update_fields=['active'])
+            return True
+
+    def _load_alert(self, alert_id):
+        try:
+            return CommCareHQAlert.objects.get(
+                created_by_domain=self.domain,
+                id=alert_id
+            )
+        except CommCareHQAlert.DoesNotExist:
+            return None
```

Comment:
The functions that call this will simply do nothing if this returns `None`, which seems like an unexpected outcome from the user's perspective. What do you think of communicating this error the the user?

Reply:
One genuine possible edge case is, if a user loads the page and tries to update an alert that is deleted by another user in the interim.
I can update the error message to say "Alert not found" if that seems like worth it.

Reply:
https://github.com/dimagi/commcare-hq/pull/33592/commits/73f9a875ba16b83557c65ba5e406843a626dd86d